File: src/sdk/ts/src/experimental.ts
/**
 * ⚠️ **Experimental.** The facts / grounding API — constant grounding content
 * an agent always needs (a shop's address, a brand's voice) plus the
 * transcript-derived re-injection freshness gate (ADR-0014).
 *
 * This surface is behind the `experimental` namespace on purpose: it is new and
 * may change or be removed without a major-version bump. Reach it as
 * `experimental.FactCatalog` (`import { experimental } from "@ratel-ai/sdk"`) so
 * that dependence on an unstable API is explicit at the import site. Constructing
 * a {@link FactCatalog} also logs a one-time warning (silence it with
 * `RATEL_EXPERIMENTAL_SILENCE=1`).
 *
 * @module
 */

export type { Fact, FactHit } from "../native/index.cjs";
export type { CapabilityFactHit } from "./capabilities.js";
export type { FactCatalogOptions } from "./fact-catalog.js";
export { FactCatalog } from "./fact-catalog.js";
export type {
  FactCandidate,
  GroundingItem,
  GroundingResult,
  GroundingSnapshotItem,
  GroundOptions,
  GroundSnapshotOptions,
  InjectionDecision,
  InjectionDecisionReason,
  InjectionPolicy,
  InjectionReason,
  LedgerEntry,
  PlanInjectionInput,
} from "./grounding.js";
export {
  FACT_ID_PATTERN,
  factHash,
  groundingMarker,
  Pin,
  planInjection,
  readGroundingLedger,
} from "./grounding.js";
export { FactRegistry } from "./registry.js";

File: src/sdk/python/ratel_ai/experimental.py
"""Experimental facts/grounding API — opt in explicitly.

This module re-exports the entire facts and grounding surface: `FactCatalog` /
`Fact` (the push-path grounding analogue — constant content injected into the
context), the pure `plan_injection` freshness gate, and their supporting types.

It is an **experimental** API. It may change or be removed without a major
version bump, so it lives outside the stable root package instead of `ratel_ai`.
Import from here so the dependency is explicit:

    from ratel_ai.experimental import FactCatalog, Fact, plan_injection

Constructing a `FactCatalog` emits a one-time `ExperimentalWarning`; set
`RATEL_EXPERIMENTAL_SILENCE=1` to silence it.
"""

from __future__ import annotations

from .fact_catalog import (
    ExperimentalWarning,
    Fact,
    FactCatalog,
    FactHit,
    FactRegistry,
    Pin,
)
from .grounding import (
    FACT_ID_PATTERN,
    FactCandidate,
    GroundingItem,
    GroundingResult,
    GroundOptions,
    InjectionDecision,
    InjectionDecisionReason,
    InjectionPolicy,
    InjectionReason,
    LedgerEntry,
    PinTier,
    fact_hash,
    grounding_marker,
    plan_injection,
    read_grounding_ledger,
)

__all__ = [
    "FACT_ID_PATTERN",
    "ExperimentalWarning",
    "Fact",
    "FactCandidate",
    "FactCatalog",
    "FactHit",
    "FactRegistry",
    "GroundOptions",
    "GroundingItem",
    "GroundingResult",
    "InjectionDecision",
    "InjectionDecisionReason",
    "InjectionPolicy",
    "InjectionReason",
    "LedgerEntry",
    "Pin",
    "PinTier",
    "fact_hash",
    "grounding_marker",
    "plan_injection",
    "read_grounding_ledger",
]
